timemachine/fe/single_topology_v3.py

```diff
@@ -0,0 +1,539 @@
+import warnings
+from collections.abc import Iterable
+
+import numpy as np
+
+from timemachine.fe import system, topology, utils
+from timemachine.fe.dummy import identify_dummy_groups, identify_root_anchors
+from timemachine.lib import potentials
+
+
+class MultipleAnchorWarning(UserWarning):
+    pass
+
+
+class CoreBondChangeWarning(UserWarning):
+    pass
+
+
+class MissingAngleError(RuntimeError):
+    pass
+
+
+def get_romol_bonds(mol):
+    """
+    Return bond idxs given a mol. These are not canonicalized.
+    """
+    bond_list = []
+    for bond in mol.GetBonds():
+        bond_list.append([bond.GetBeginAtomIdx(), bond.GetEndAtomIdx()])
+    return bond_list
+
+
+def get_neighbors(bond_idxs, query):
+    """
+    Get neighbors of query atom query.
+    """
+    nbs = set()
+    for i, j in bond_idxs:
+        if i == query:
+            nbs.add(j)
+        if j == query:
+            nbs.add(i)
+    return nbs
+
+
+def recursive_map(items, mapping):
+    """recursively replace items in a list of tuple
+    mapping = np.arange(100)[::-1]
+    items = [[0,2,3], [5,1,[2,5,6]], 3]
+    result = recursive_map(items, mapping)
+    """
+    if isinstance(items, Iterable):
+        res = []
+        for item in items:
+            res.append(recursive_map(item, mapping))
+        return tuple(res)
+    else:
+        return mapping[items]
+
+
+def find_nbs(mol, atom):
+    nbs = [a.GetIdx() for a in mol.GetAtomWithIdx(atom).GetNeighbors()]
```

Comment:
nit: annotation 

(return type differs from `get_neighbors`)

```suggestion
def find_nbs(mol, atom_idx: int) -> List[Int]:
    nbs = [a.GetIdx() for a in mol.GetAtomWithIdx(atom_idx).GetNeighbors()]
```

Reply:
hm yeah - I don't think we have doubly connected atoms, so either should be fine.